lang/attribute/program/src/declare_program/mods/internal.rs

```diff
@@ -171,7 +171,7 @@ fn gen_internal_accounts_common(
             let accs_struct = syn::parse2(accs_struct).expect("Failed to parse as syn::ItemStruct");
             let accs_struct =
                 accounts::parse(&accs_struct).expect("Failed to parse accounts struct");
-            gen_accounts(&accs_struct)
+            gen_accounts(&accs_struct, quote!(super::__ID))
```

Comment:
For the sake of consistency, it would be better to use [`get_canonical_program_id`](https://github.com/coral-xyz/anchor/blob/518b73f21959efc86407ab1da03f538ae6c53121/lang/attribute/program/src/declare_program/common.rs#L10) function rather than hardcoding this.